packages/core/src/utils-hoist/envelope.ts

```diff
@@ -259,7 +259,7 @@ export function createEventEnvelopeHeaders(
     ...(sdkInfo && { sdk: sdkInfo }),
     ...(!!tunnel && dsn && { dsn: dsnToString(dsn) }),
     ...(dynamicSamplingContext && {
-      trace: dropUndefinedKeys({ ...dynamicSamplingContext }),
+      trace: dynamicSamplingContext,
```

Comment:
I don't think we have to do this here, undefined stuff is dropped when it is JSON-stringified anyhow 🤔